spring-pulsar-spring-boot-autoconfigure/src/main/java/org/springframework/pulsar/autoconfigure/PulsarProperties.java

```diff
@@ -2002,10 +2002,10 @@ public Map<String, Object> buildProperties() {
 			map.from(this::getTlsTrustStorePassword).to(properties.in("tlsTrustStorePassword"));
 			map.from(this::getTlsCiphers).to(properties.in("tlsCiphers"));
 			map.from(this::getTlsProtocols).to(properties.in("tlsProtocols"));
-			map.from(this::getConnectTimeout).to(properties.in("connectTimeout"));
-			map.from(this::getReadTimeout).to(properties.in("readTimeout"));
-			map.from(this::getRequestTimeout).to(properties.in("requestTimeout"));
-			map.from(this::getAutoCertRefreshTime).to(properties.in("autoCertRefreshTime"));
+			map.from(this::getConnectionTimeout).asInt(Duration::toMillis).to(properties.in("connectionTimeoutMs"));
```

Comment:
The ConfigurationDataUtils does not support Duration at all. Eventhough the PulsarAdminstration.init mucks w/ the props (removes them) I think turning them into `ms` values like the other time props are all handled makes it consistent .

Reply:
Yes !
And in 2.12, these properties will be handled by CCD: https://github.com/apache/pulsar/pull/17375
The names will be `connectionTimeoutMs`, `requestTimeoutMs`, `readTimeoutMs` and `autoCertRefreshSeconds`

Reply:
Awesome! Today I was going to look if there was an issue for that already and if not, contribute. Good to hear this.